### PR TITLE
Support a UBI-only mode in the operator

### PR DESCRIPTION
**Pull request by pebrc**
Add a new operator flag to force the operator to only manage containers based on UBI stack images. 

* Replace container-suffix with ubi-only. 
* Auto-suffix default container images with `-ubi8`. 
* Validate a minimum version for UBI images.

### CODE REVIEW

File: cmd/manager/main.go [hunk 1]
```diff
@@ -217,6 +213,11 @@ func Command() *cobra.Command {
 		"",
 		"Kubernetes namespace the operator runs in",
 	)
+	cmd.Flags().Bool(
+		operator.UBIOnlyFlag,
+		false,
+		"All Docker images run by the operator must be based on a UBI image (available as of 7.10.0)",
```
Comment (charith-elastic) [inline cmd/manager/main.go:219]:
```suggestion
		"Use only UBI container images to deploy Elastic Stack applications. UBI images are only available from 7.10.0 onward.",
```

File: cmd/manager/main.go [hunk 2]
```diff
@@ -374,11 +375,11 @@ func startOperator(stopChan <-chan struct{}) error {
...
-	// set a custom container suffix if specified
-	containerSuffix := viper.GetString(operator.ContainerSuffixFlag)
-	if containerSuffix != "" {
-		log.Info("Setting default container suffix", "container_suffix", containerSuffix)
-		container.SetContainerSuffix(containerSuffix)
+	// enforce UBI stack images if requested
+	ubiOnly := viper.GetBool(operator.UBIOnlyFlag)
+	if ubiOnly {
+		container.SetContainerSuffix("-ubi8")
+		version.GlobalMinStackVersion = version.MustParse("7.10.0")
```
Comment (charith-elastic) [inline cmd/manager/main.go:382]:
```suggestion
		version.GlobalMinStackVersion = version.From(7,10,0)
```

File: cmd/manager/main.go [hunk 3]
```diff
@@ -510,7 +511,7 @@ func startOperator(stopChan <-chan struct{}) error {
 	go asyncTasks(mgr, cfg, managedNamespaces, operatorNamespace, string(operatorInfo.OperatorUUID))
 
 	log.Info("Starting the manager", "uuid", operatorInfo.OperatorUUID,
-		"namespace", operatorNamespace, "version", operatorInfo.BuildInfo.Version,
+		"namespace", operatorNamespace, "esversion", operatorInfo.BuildInfo.Version,
```
Comment (charith-elastic) [inline cmd/manager/main.go:514]: Is this intentional?

```suggestion
		"namespace", operatorNamespace, "version", operatorInfo.BuildInfo.Version,
```

File: docs/operating-eck/operator-config.asciidoc [hunk 4]
```diff
@@ -31,6 +31,7 @@ ECK can be configured using either command line flags or environment variables.
 |metrics-port |0 |Prometheus metrics port. Set to 0 to disable the metrics endpoint.
 |namespaces |"" |Namespaces in which this operator should manage resources. Accepts multiple comma-separated values. Defaults to all namespaces if empty or unspecified.
 |operator-namespace |"" |Namespace the operator runs in. Required.
+|ubi-only | false | All Docker images run by the operator must be based on a UBI image (available as of 7.10.0)
```
Comment (charith-elastic) [inline docs/operating-eck/operator-config.asciidoc:34]:
```suggestion
|ubi-only | false | Use only UBI container images to deploy Elastic Stack applications. UBI images are only available from 7.10.0 onward.
```

File: cmd/manager/main.go (hunk 3 above)
Comment (pebrc) [reply cmd/manager/main.go:514]: Overagressive IntelliJ refactoring :-(

**Review by charith-elastic — APPROVED** [review]
LGTM. Min Stack version validation and image suffix seem to work as expected. Didn't test the UBI images because they are not available yet.

**Comment by pebrc** [comment]
@charith-elastic  you can test with `7.10.0-SNAPSHOT` versions at least for some of the stack applications.

**Comment by charith-elastic** [comment]
Tested Elasticsearch, Kibana and APM Server :+1:

**Pull request merged**